src/extension_instance.ml

```diff
@@ -125,6 +125,63 @@ let check_ocaml_lsp_available sandbox =
           current sandbox.")
 ;;
 
+let suggest_to_run_dune_pkg_lock t =
+  let open Promise.Syntax in
+  let (_ : unit Promise.t) =
+    let+ maybe_choice =
+      Window.showWarningMessage
+        ~message:
+          "Dune Package Manager is detected but no lock file is present. Do you want to \
+           run `dune pkg lock`"
+        ~choices:
+          [ ( "Run dune pkg lock"
+            , fun () ->
+                let _ =
+                  let options =
+                    ProgressOptions.create
+                      ~location:(`ProgressLocation Notification)
+                      ~title:"Running `dune pkg lock`"
+                      ~cancellable:false
+                      ()
+                  in
+                  let task ~progress:_ ~token:_ =
+                    let+ result =
+                      let _ =
+                        let terminal = Terminal_sandbox.create t.sandbox in
+                        let _ = Terminal_sandbox.show ~preserveFocus:true terminal in
+                        Terminal_sandbox.send terminal "dune pkg lock"
+                      in
+                      Ok () |> Promise.return
+                    in
+                    match result with
+                    | Ok () -> Ojs.null
+                    | Error err ->
+                      show_message
+                        `Error
+                        "An error occured while running dune pkg lock: %s"
+                        err;
+                      Ojs.null
```

Comment:
`result` is always `Ok ()` so this never happen.

And I mean, this is a shame, we should be able to wait for the command to finish and actually check the result, isn't there a way to do that ? I think this is what happen when installing a package.